pkg/cert/v1beta1/generate/generate.go

```diff
@@ -0,0 +1,263 @@
+		return err
+	}
+
+	// Create secret with CA cert and server cert/key.
+	// Add ownerReferences to clean-up secret with cert generator Job.
+	jobUID := certGeneratorJob.UID
+	webhookCertSecret := &corev1.Secret{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       "Secret",
+			APIVersion: "v1",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      consts.Secret,
+			Namespace: o.namespace,
+			OwnerReferences: []metav1.OwnerReference{
+				{
+					APIVersion: "batch/v1",
+					Kind:       "Job",
+					Controller: pointer.BoolPtr(true),
+					Name:       consts.JobName,
+					UID:        jobUID,
+				},
+			},
+		},
+		Type: corev1.SecretTypeTLS,
+		Data: map[string][]byte{
+			"ca.key":  caKeyPair.keyPem,
+			"ca.crt":  caKeyPair.certPem,
+			"tls.key": keyPair.keyPem,
+			"tls.crt": keyPair.certPem,
+		},
+	}
+
+	oldSecret := &corev1.Secret{}
+	err := kubeClient.GetResources(ctx, client.ObjectKey{Namespace: o.namespace, Name: consts.Secret}, oldSecret)
+	switch {
+	case err != nil && !k8serrors.IsNotFound(err):
+		return err
+	case err == nil:
+		klog.Warning("Previous secret was found and removed.")
+		if err = kubeClient.DeleteResources(ctx, oldSecret); err != nil {
+			return err
+		}
+	}
+
+	klog.Infof("Creating Secret: %s", consts.Secret)
+	if err = kubeClient.CreateResources(ctx, webhookCertSecret); err != nil {
+		return err
+	}
+	return nil
+}
+
+// injectCert apply patch to ValidatingWebhookConfiguration and MutatingWebhookConfiguration
+func (o *generateOptions) injectCert(ctx context.Context, kubeClient *kube.Client, caKeypair *certificates) error {
+	validatingConf := &admissionregistrationv1.ValidatingWebhookConfiguration{}
+	if err := kubeClient.GetResources(ctx, client.ObjectKey{Name: consts.Webhook}, validatingConf); err != nil {
+		return err
+	}
+	newValidatingConf := validatingConf.DeepCopy()
+	newValidatingConf.Webhooks[0].ClientConfig.CABundle = caKeypair.certPem
+
+	klog.Info("Trying to patch ValidatingWebhookConfiguration adding the caBundle.")
+	for i := 0; i <= 5; i++ {
+		err := kubeClient.PatchResources(ctx, newValidatingConf, client.MergeFrom(validatingConf))
+		switch {
+		case err == nil:
+			break
+		case i == 5 && err != nil:
+			klog.Errorf("Unable to patch ValidatingWebhookConfiguration %s", consts.Webhook)
+			return err
+		default:
+			klog.Warning("Webhook are not patched. Retrying in 5s...")
+			time.Sleep(time.Second * 5)
+		}
+	}
+
+	mutatingConf := &admissionregistrationv1.MutatingWebhookConfiguration{}
+	if err := kubeClient.GetResources(ctx, client.ObjectKey{Name: consts.Webhook}, mutatingConf); err != nil {
+		return err
+	}
+	newMutatingConf := mutatingConf.DeepCopy()
+	newMutatingConf.Webhooks[0].ClientConfig.CABundle = caKeypair.certPem
+	newMutatingConf.Webhooks[1].ClientConfig.CABundle = caKeypair.certPem
+
+	klog.Info("Trying to patch MutatingWebhookConfiguration adding the caBundle.")
+	for i := 0; i <= 5; i++ {
```

Comment:
Can we avoid magic numbers here? Maybe we can define it as a constant

Reply:
Thanks, I fixed it.